WireUI/Sources/WireMoveToFolderUI/Views/CreateFolder/CreateFolder.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireDesign
+import WireReusableUIComponents
+
+public struct CreateFolder: View {
+
+    // MARK: - Properties
+
+    @Environment(\.dismiss) private var dismiss
+    @ObservedObject private var viewModel: CreateFolderViewModel
+    private let conversationName: String
+    private let onFolderCreated: (Folder) -> Void
+
+    public init(
+        viewModel: CreateFolderViewModel,
+        conversationName: String,
+        onFolderCreated: @escaping (Folder) -> Void
+    ) {
+        self.viewModel = viewModel
+        self.conversationName = conversationName
+        self.onFolderCreated = onFolderCreated
+    }
+
+    public var body: some View {
+        NavigationStack {
```

Comment:
Personally I think the `NavigationStack` should be applied in `FolderPickerHostingController`. Doing it here limits where we can use `CreateFolder` from.

Reply:
After thinking about it, I believe keeping the NavigationStack in the SwiftUI view is the better approach for this implementation. Here's why:

1. The current architecture has a clean flow:
   UIKit -> FolderPickerBuilder -> FolderPickerHostingController -> FolderPicker (SwiftUI) -> Create Folder (SwiftUI)
   
   The hosting controller serves as a bridge between UIKit and SwiftUI, while letting SwiftUI handle its own navigation concerns.

2. The SwiftUI view (FolderPicker) uses SwiftUI-specific navigation patterns:
   - NavigationLink to CreateFolder view
   - Environment values like @Environment(\.dismiss)
   - Toolbar items and navigation bar configuration
   
   Moving the NavigationStack to the hosting controller would make these harder to manage.

3. The current implementation maintains better state management where all SwiftUI-related state and navigation stays within the SwiftUI hierarchy.